lib/metric-config-parser/metric_config_parser/tests/test_analysis.py

```diff
@@ -763,3 +765,68 @@ def test_multiple_nesting(self, experiments, config_collection):
         assert len(metric.depends_on[0].metric.depends_on) == 1
         assert metric.depends_on[0].metric.depends_on[0].metric.name == "spam"
         assert metric.depends_on[0].metric.depends_on[0].metric.select_expression == "1"
+
+    def test_default_aggregation_units(self, experiments, config_collection):
+        config_str = dedent(
+            """
+            [metrics]
+            weekly = ["active_hours"]
+
+            [metrics.active_hours.statistics.bootstrap_mean]
+            """
+        )
+
+        spec = AnalysisSpec.from_dict(toml.loads(config_str))
+        cfg = spec.resolve(experiments[0], config_collection)
+
+        metric = [m for m in cfg.metrics[AnalysisPeriod.WEEK] if m.metric.name == "active_hours"][
+            0
+        ].metric
+        assert metric.aggregation_units == [AnalysisUnit.CLIENT]
+
+        data_source = metric.data_source
+        assert data_source.aggregation_units == [AnalysisUnit.CLIENT]
+
+    @pytest.mark.parametrize(
```

Comment:
Excellent use of parametrize here. This test checks all the invalid combinations, would it be useful or unneccessary to have a similar test to test the valid combinations?

Reply:
Sure, can't hurt.